pkg/status/aggregation/daemonset.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2025 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aggregation
+
+func AggregateDaemonSetStatus(statuses []map[string]interface{}) (map[string]interface{}, error) {
+	if len(statuses) == 0 {
+		return nil, nil
+	}
+
+	if len(statuses) == 1 {
+		return statuses[0], nil
+	}
```

Comment:
This code is going to be repeated for every kind of object, right? Wouldn't it be better for the caller to handle this case once and for all?